core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -715,14 +715,17 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
     }
   }
 
-  def updateIntersectionGroupMembers(groupsToIntersect: Seq[GroupsToIntersect], dataAccess:DataAccess): ReadWriteAction[Iterable[RawlsGroupRef]] = {
```

Comment:
The old version of this function would calculate the intersection and overwrite members one bye one for each group. This involved flattening each group (a recursive call) and usually the same group was involved in each intersection so it was flattened repeatedly.

The new version flattens all groups once and only once then uses that to calculate the appropriate intersections then executes one series of calls to overwrite the group memberships.